airbyte-integrations/connectors/source-gcs/source_gcs/source.py

```diff
@@ -0,0 +1,71 @@
+#
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+#
+
+
+import json
+from datetime import datetime
+from typing import Dict, Generator
+
+from airbyte_cdk.logger import AirbyteLogger
+from airbyte_cdk.models import (
+    AirbyteCatalog,
+    AirbyteConnectionStatus,
+    AirbyteMessage,
+    AirbyteRecordMessage,
+    AirbyteStream,
+    ConfiguredAirbyteCatalog,
+    Status,
+    Type,
+)
+from airbyte_cdk.sources import Source
+
+from .helpers import construct_file_schema, get_gcs_blobs, read_csv_file, get_stream_name
+MIN_BYTES_READ_GENERATE_SCHEMA = 102400
+
+
+class SourceGCS(Source):
+    def check(self, logger: AirbyteLogger, config: json) -> AirbyteConnectionStatus:
+        """
+        Check to see if a client can be created and list the files in the bucket.
+        """
+        try:
+            blobs = get_gcs_blobs(config)
+            if not blobs:
+                return AirbyteConnectionStatus(status=Status.FAILED, message="No compatible file found in bucket")
+            return AirbyteConnectionStatus(status=Status.SUCCEEDED)
+        except Exception as e:
+            return AirbyteConnectionStatus(status=Status.FAILED, message=f"An exception occurred: {str(e)}")
+
+    def discover(self, logger: AirbyteLogger, config: json) -> AirbyteCatalog:
+        streams = []
+
+        blobs = get_gcs_blobs(config)
+        for blob in blobs:
+            # Read the first 0.1MB of the file to determine schema
+            df = read_csv_file(
+                blob, limit_bytes=MIN_BYTES_READ_GENERATE_SCHEMA)
+            stream_name = get_stream_name(blob)
+            json_schema = construct_file_schema(df)
+            streams.append(AirbyteStream(
+                name=stream_name, json_schema=json_schema, supported_sync_modes=["full_refresh"]))
+
+        return AirbyteCatalog(streams=streams)
+
+    def read(
+        self, logger: AirbyteLogger, config: json, catalog: ConfiguredAirbyteCatalog, state: Dict[str, any]
```

Comment:
you're not using catalog at all! this means a user won't be able to turn off streams he doesn't need. Please fix this

Reply:
@davydov-d Do you have an example of a catalog implementation for an undefined source like this?

Reply:
I have a json representation of a sample configured catalog which looks as follows
```
{
  "streams": [
    {
      "stream": {
        "name": "cadence_memberships",
        "json_schema": {},
        "supported_sync_modes": ["full_refresh", "incremental"],
        "source_defined_cursor": true,
        "default_cursor_field": ["updated_at"],
        "source_defined_primary_key": [["id"]]
      },
      "sync_mode": "incremental",
      "destination_sync_mode": "overwrite",
      "cursor_field": ["updated_at"]
    }
  ]
}
```

Also you can look how the catalog is treated in the CDK's [read](https://github.com/airbytehq/airbyte/blob/master/airbyte-cdk/python/airbyte_cdk/sources/abstract_source.py#L107) method.

Reply:
I'm still not sure what you are suggesting. Is it not enough? https://github.com/airbytehq/airbyte/blob/22bf955d5bd17a9a405372a7df50267a23a4961d/airbyte-integrations/connectors/source-gcs/integration_tests/configured_catalog.json

Reply:
That makes sense. Thank you @davydov-d, I've implemented the logic now.